test/test.c

```diff
@@ -7430,14 +7430,15 @@ _checkPool(natsConnection *nc, char **expectedURLs, int expectedURLsCount)
     bool    ok;
 
     natsMutex_Lock(nc->mu);
-    if (nc->srvPool->size != expectedURLsCount)
```

Comment:
You could still have this "if/print error" statement at the end (before releasing the mutex) so that we have this printed in case it helps.
```
    ...
    if (nc->srvPool->size != expectedURLsCount)
        printf("Expected pool size to be %d, got %d\n", expectedURLsCount, nc->srvPool->size);

    natsMutex_Unlock(nc->mu);
    return NATS_ERR;
}
```

Reply:
@kozlovic this seemed like a 1-liner to add, but upon further reading the function, I don't think the retry logic was quite correct, so I refactored the function.